### DISCUSSION TITLE
Extend Directus rollup.config for custom extensions

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#6599] [Feature Requests]
As proposed in https://github.com/directus/directus/issues/6556, devs should be able to extend the Directus' default rollup config, to be able to use custom modules like Sass, Less or any other [awesome stuff.](https://github.com/rollup/awesome)

A great example of how it could be done is [Next.js](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config) IMHO.

**Comment by nickluger** [comment]
I just saw that @nickrum used `rollup-plugin-styles` which should already make all kind of CSS extensions work.

**Comment by Guiqft** [comment]
Hi, I was able to use scss in my interface on rc.83 with the following [rollup.config.js](https://github.com/Guiqft/directus-interface-template/blob/master/rollup.config.js)

**Comment by nickrum** [comment]
Loading a config file and using it to overwrite rollup options is actually the easy part. Designing a good config system is unfortunately not as straightforward.
We wouldn't want users to overwrite settings that we specifically set and changing them wouldn't make sense within the context of Directus extensions.
Also, extending rollup's `plugins` config is particularly tricky as the order of those plugins matters. We'd have to provide ways to inject own rollup plugins at certain points within the plugin chain and do it in such a way that we have the flexibility to add, remove or swap out the "built-in" plugins in the future.

The question is whether we should support this or if advanced use cases have to be handled by the user by using rollup directly.

**Reply by tylerforesthauser** [reply]
I definitely think for advanced use cases, it should be up to the user to provide their own rollup config. What would be handy would be providing the core config (or equivalent) in the docs for easy reference.

**Reply by nickluger** [reply]
Agree 100% with @nickrum, and @tylerforesthauser (regarding docs on how the custom file must look like to work with Directus).

**Comment by nickrum** [comment]
Support for a very basic config file to add custom rollup plugins to the build chain has been added in #7332.
